config/config_loader.go

```diff
@@ -33,26 +33,35 @@ import (
 var (
 	consumerConfig *ConsumerConfig
 	providerConfig *ProviderConfig
+	routerConfig   *ConditionRouterConfig
 	maxWait        = 3
 )
 
 // loaded comsumer & provider config from xxx.yml, and log config from xxx.xml
 // Namely: dubbo.comsumer.xml & dubbo.provider.xml in java dubbo
 func init() {
 	var (
-		confConFile, confProFile string
+		confConFile, confProFile, confRouterFile string
 	)
 
 	confConFile = os.Getenv(constant.CONF_CONSUMER_FILE_PATH)
 	confProFile = os.Getenv(constant.CONF_PROVIDER_FILE_PATH)
+	confRouterFile = os.Getenv(constant.CONF_ROUTER_FILE_PATH)
+
 	if errCon := ConsumerInit(confConFile); errCon != nil {
 		log.Printf("[consumerInit] %#v", errCon)
 		consumerConfig = nil
 	}
+
 	if errPro := ProviderInit(confProFile); errPro != nil {
 		log.Printf("[providerInit] %#v", errPro)
 		providerConfig = nil
 	}
+
+	if errPro := RouterInit(confRouterFile); errPro != nil {
```

Comment:
You had initialized the routerConfig here, why do you try to initialize the routerConfig in initRouterUrl() method? I think initializing routerConfig here is enough? What do you think?

Reply:
agree